src/pkt_handlers.c

```diff
@@ -1347,6 +1347,7 @@ void counters_handler(struct channels_list_entry *chptr, struct packet_ptrs *ppt
 
   if (pptrs->l3_proto == ETHERTYPE_IP) pdata->pkt_len = ntohs(((struct pm_iphdr *) pptrs->iph_ptr)->ip_len);
   else if (pptrs->l3_proto == ETHERTYPE_IPV6) pdata->pkt_len = ntohs(((struct ip6_hdr *) pptrs->iph_ptr)->ip6_plen)+IP6HdrSz;
+  else if (config.aggregate_unknown_etype) pdata->pkt_len = pptrs->pkthdr->len-ETHER_HDRLEN;
```

Comment:
I considered setting the unknown Ethernet frame "packet length" to `pptrs->caplen` instead of `pptrs->len` (less the Ethernet header length). I'm not sure which is the best choice here, but I decided on the original packet length since we're throwing away the payload on aggregation anyway, and it's probably more useful to the user to have the actual packet length in the aggregation purge events...